articles/prototyping-machine-learning-models-with-streamlit/index.md

```diff
@@ -0,0 +1,185 @@
+
+#### Introduction to Streamlit
+Streamlit is an Open source python framework that enables developers to demo or prototype their ML models and a lot more on a web application effortlessly, you can simply turn your data science scripts into a website with few lines of code. It saves time, energy, and is also simple to learn, use, and fast too. Now we’ll dive straight into how to install and setup streamlit and then build a text summarizer web app.
+
+##### Streamlit Set up
+The installation is very simple, just run this on your command prompt or terminal
+
+```python
+Pip install streamlit
+Streamlit hello
+```
+To import it, use the simple
+
+```python
+import pandas as pd
+import streamlit as st
+import plotly.express as px
+```
+And to run your streamlit app simply use
+
+```python
+Streamlit run app.py
+```
+#### Building Sentiment Analyzer
+Now for this introductory tutorial, we will be building a sentiment analyzer model and prototype it with streamlit. The first step will be getting a Dataset, and the dataset we will be using can be found HERE, next let’s create a directory and create a virtual environment.
+
+```python
+$ mkdir textanalyzer
+$ cd textanalyzer
+```
+Now to create the Virtual environment use the code below
+
+```python
+$ python3.8 -m venv env
+$ source env/bin/activate
+```
+Now copy the dataset as it is in the folder and paste it into this textanalyzer directory, now create a file and save it as sentiment_analyzer, this will be where we will be writing our code to build the streamlit app. The next thing will be to import all the packages we will be using for this tutorial.
+
+```python
+import streamlit as st
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.naive_bayes import BernoulliNB
+from sklearn.metrics import confusion_matrix
+import numpy as np
+import itertools
+import matplotlib.pyplot as plt
+```
+After that let’s add a header for our web app and a subheader and to do that we simply use the code below
+
+```python
+st.title("Sentiment Analyzer Based On Text Analysis ")
+st.subheader("Paras Patidar - MLAIT")
+st.write('\n\n')
+```
+The next step will be to write a function that will get all the data from our directory and bring them up,
+
+```python
+@st.cache
+def get_all_data():
+    root = "Datasets/"
+    with open(root + "imdb_labelled.txt", "r") as text_file:
+        data = text_file.read().split('\n')
+         
+    with open(root + "amazon_cells_labelled.txt", "r") as text_file:
+        data += text_file.read().split('\n')
+
+    with open(root + "yelp_labelled.txt", "r") as text_file:
+        data += text_file.read().split('\n')
+
+    return data
+```
+If you notice in the function above the first line is *@st.cache*, this mark tells streamlit that whenever the function is called that there are some certain requirements it should check for and they should be met, now create a variable called all_data which should call the *get_all_data* function we just wrote.
```

Comment:
What are the certain requirements you are referring to? Why is it needed?

Reply:
the requirements are things that have to do with the compiler and not the developer and we don't need it, it a more like a mark to the compiler.